app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.wife = None
+        self.husband = None
+
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = [Person(data["name"], data["age"]) for data in people]
+
+    for data in people:
+        person = Person.people[data["name"]]
+        spouse_name = data.get("wife") or data.get("husband")
```

Comment:
The code assumes that each person can have either a 'wife' or 'husband' key. Ensure that the input data is consistent with this assumption to avoid potential issues.